### ISSUE TITLE
Setting <AnalysisMode<Category>> to Default doesn't work if <AnalysisMode> is All

### ISSUE THREAD

**Issue by jonathanou** [dotnet/roslyn-analyzers#6987]
### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.402](https://dotnet.microsoft.com/download/dotnet/7.0)

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview.23420.2

### Describe the bug

In my csproj file, if I do this:

```xml
<AnalysisMode>All</AnalysisMode>
<AnalysisModePerformance>None</AnalysisModePerformance>
```
Then all rules are enabled except for performance rules, as expected.

But if I set the performance rules to `Default`:

```xml
<AnalysisMode>All</AnalysisMode>
<AnalysisModePerformance>Default</AnalysisModePerformance>
```

*All* performance rules are enabled, even though not every performance rule is enabled by default.

The only category that works as expected is `AnalysisModeStyle`.  Setting `AnalysisModeStyle` to `Default` does indeed only enable default style rules.

### Steps To Reproduce

1.  Create a console app csproj and add the following code:

```cs
string test = "Hello";
if (test.Contains("o", StringComparison.Ordinal))
{
    Console.WriteLine("Hello, World!");
}
```
No warnings are emitted.

2.  Update csproj file with `<AnalysisMode>All</AnalysisMode>`.  Notice the `test.Contains` call now emits CA 1847.
3.  Update csproj file with an additional `<AnalysisModePerformance>None<AnalysisModePerformance>`.  Notice CA 1847 goes away.
4.  Now update `<AnalysisModePerformance>` to `Default`.  CA 1847 still fires.
5.  Remove `<AnalysisMode>All</AnalysisMode>`, no warnings emitted (we go back to the state of the first step).

### Expected behavior

When setting `<AnalysisModePerformance>Default</AnalysisModePerformance>`, CA 1847 does not fire.

### Actual behavior

When setting `<AnalysisModePerformance>Default</AnalysisModePerformance>`, CA 1847 still fires.

## Additional context

From the binlog and looking at Microsoft.CodeAnalysis.NetAnalyzers.targets, I believe the problem lies in how global config files are aggregated.

When `<AnalysisMode>` is `All`, `analysislevel_6_all.globalconfig` is added to the `EditorConfigFiles` item.  `analysislevel_6_all.globalconfig` has a `global_level` of -100, and sets all rules to `warning` as expected.

Next, `<AnalysisModePerformance>` is evaluated, and `analysislevelperformance_6_default.globalconfig` is added, where the `global_level` is set to -99, allowing rules specified in this config to take precedence over `analysislevel_6_all.globalconfig`.  However, only a few rules are set to `none`, and CA 1847 is not one of them.

In the case that `<AnalysisModePerformance>` is `None`, `analysislevelperformance_6_none.globalconfig` is added and all performance rules, including CA 1847, is set to `none`, which explains why it works as expected.

I also looked at some of the other default config files, and at least for documentation and security categories, the default global config has *zero* rules specified, so the problem exists in other categories too.

The only category unaffected is Style (IDExxxx rules).  This is because `analysislevel_6_all.globalconfig` does not mention IDE rules whatsoever, leaving IDE rules as default even when `<AnalysisMode>` is `All` (which is probably a different bug?).

**Comment by mavasani** [comment]
> The only category unaffected is Style (IDExxxx rules). This is because analysislevel_6_all.globalconfig does not mention IDE rules whatsoever. So while analysislevelstyle_default.globalconfig also has zero rules specified, it ends up working as intended.

That should handled in a separate targets file generated here: https://github.com/dotnet/roslyn/blob/main/src/CodeStyle/Tools/Program.cs and packaged into the CodeStyle NuGet package and .NET SDK separately. Its likely that part is busted and hence doesn't work anymore. Can you please file a separate issue on dotnet/roslyn repo for that? Thanks!

**Comment by jonathanou** [comment]
> Can you please file a separate issue on dotnet/roslyn repo for that?

"analysislevel**style**_all.globalconfig" does correctly have all IDE rules set to warning.  It's `analysislevel_6_all.globalconfig` (the globalconfig that for all rules across all categories) that doesn't have IDE rules specified.  So I think the CodeStyle generator program is doing the right thing (if I'm understanding all of this correctly).

For reference, this is the `analysislevelstyle_all.globalconfig` on my machine from the .NET 7 SDK:

![image](https://github.com/dotnet/roslyn-analyzers/assets/1291962/f5df0cf0-91ac-4df9-ae9e-62bea6e89be8)

And this is `analysislevel_6_all.globalconfig` (it's the same thing for versions 7 and 8):

![image](https://github.com/dotnet/roslyn-analyzers/assets/1291962/a839445c-ecc3-491c-8c55-3baba61f915d)

**Comment by jonathanou** [comment]
Looking at the targets file some more, `<AnalysisModeStyle>` inherits the value of `<AnalysisMode>` if not set (from `C:\Program Files\dotnet\sdk\7.0.402\Sdks\Microsoft.NET.Sdk\codestyle\cs\build\Microsoft.CodeAnalysis.CSharp.CodeStyle.targets`):

```xml
<!-- Default 'AnalysisModeStyle' to the core 'AnalysisMode' -->
<AnalysisModeStyle Condition="'$(AnalysisModeStyle)' == ''">$(AnalysisMode)</AnalysisModeStyle>
```

It doesn't appear the other category's analysis mode does this, as I didn't see any similar logic in `Microsoft.CodeAnalysis.NetAnalyzers.targets`.

So I think one easy fix might be to stop adding `analysislevel_6_all.globalconfig` to `EditorConfigFiles`.  Instead, let each category's analysis mode inherit `<AnalysisMode>` if not set, and add each category's `globalconfig` file based on the analysis mode level to `EditorConfigFiles`.
